server/internal/router/router.go

```diff
@@ -0,0 +1,192 @@
+package router
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	v1 "github.com/llm-operator/inference-manager/api/v1"
+	"github.com/llm-operator/inference-manager/server/internal/config"
+	"github.com/llm-operator/inference-manager/server/internal/k8s"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"k8s.io/apimachinery/pkg/util/runtime"
+)
+
+const tickPeriod = 60 * time.Second
+
+type model struct {
+	id string
+}
+
+// R manages the routing table of models to engines.
+type R struct {
+	// m is the route map.
+	m             *routeMap
+	engineClients map[string]v1.InferenceEngineInternalServiceClient
+	imeConfig     config.InferenceManagerEngineConfig
+}
+
+// New creates a new router.
+func New(c config.InferenceManagerEngineConfig) *R {
+	return &R{
+		m:             newRouteMap(),
+		engineClients: make(map[string]v1.InferenceEngineInternalServiceClient),
+		imeConfig:     c,
+	}
+}
+
+// Run updates the routes periodically and upon pod events.
+func (r *R) Run(ctx context.Context, errCh chan error) error {
+	coreClient, err := k8s.NewCoreClient()
```

Comment:
How about adding another parameter `New`  for this? As this is an immutable field that is consistently used in this struct, we can add a field to the struct and reuse, instead of passing to `refreshRoutes`.

Reply:
Passing it to `refreshRoutes` is for making mock test easily, since `Infromer` need `coreClientSet`, but `refreshRoutes` need `listPods` only

Reply:
I'm not thinking that mocking might not be hard if we use the fake k8s client, but I could be wrong. We can revisit later.